client.go

```diff
@@ -316,7 +317,9 @@ func (c *Client) DoTimeout(req *Request, resp *Response, timeout time.Duration)
 // It is recommended obtaining req and resp via AcquireRequest
 // and AcquireResponse in performance-critical code.
 func (c *Client) DoDeadline(req *Request, resp *Response, deadline time.Time) error {
-	return clientDoDeadline(req, resp, deadline, c)
+	ctx, cancel := context.WithDeadline(context.Background(), deadline)
+	defer cancel()
+	return clientDoDeadline(req, resp, ctx, c)
```

Comment:
i suggest do not change default behavior for API, 

 possible options: use `clientDoDeadline(context.Background(), req, resp, deadline,  c)`


 new api:

```go
func (c *HostClient) DoCtx( ctx context.Context, req *Request, resp *Response) error {
	return clientDoDeadline(ctx, req, resp, c, ctx.Deadline())
}

```

Reply:
Not sure if I can see where api behavior changes?
This is an internal function which is only used in current scope and all public APIs will behave the same as it was before.

Reply:
It makes additional allocations. 

 But it private copy, probably it's not so important. 

 Also, with context deadlines `tc := acquireTimer(timeout)` logic could be removed from `clientDoDeadline`.

Reply:
> Also, with context deadlines tc := acquireTimer(timeout) logic could be removed from clientDoDeadline.

Good point, removed this.